plugin/pkg/auth/authorizer/node/graph.go

```diff
@@ -0,0 +1,244 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package node
+
+import (
+	"sync"
+
+	"k8s.io/kubernetes/pkg/api"
+	pvutil "k8s.io/kubernetes/pkg/api/persistentvolume"
+	podutil "k8s.io/kubernetes/pkg/api/pod"
+	"k8s.io/kubernetes/third_party/forked/gonum/graph"
+	"k8s.io/kubernetes/third_party/forked/gonum/graph/simple"
+)
+
+// NamedNode implements graph.Node and remembers the type, namespace, and name of it's related API object
+type NamedNode struct {
+	nodeType  nodeType
+	name      string
+	namespace string
+	id        int
+}
+
+func NewNode(nodeType nodeType, namespace, name string, id int) *NamedNode {
+	return &NamedNode{
+		nodeType:  nodeType,
+		name:      name,
+		namespace: namespace,
+		id:        id,
+	}
+}
+func (n *NamedNode) ID() int {
+	return n.id
+}
+func (n *NamedNode) String() string {
+	if len(n.namespace) == 0 {
+		return nodeTypes[n.nodeType] + ":" + n.name
+	}
+	return nodeTypes[n.nodeType] + ":" + n.namespace + "/" + n.name
+}
+
+// NodeEdge is a graph edge that is related to a kubernetes Node
+type NodeEdge struct {
+	F    graph.Node
+	T    graph.Node
+	Node graph.Node
```

Comment:
This looks bonkers, but I'm guessing you intend for this to be a `KubernetesNode` or maybe a `RootNode` or its pointing  to some important node that you want a direct link to in your edge?  Regardless, see an Edge with a Node embedded in it looks really strange, find a better name.